src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1

```diff
@@ -304,4 +310,175 @@ class WinGetSourcesResource
     }
 }
 
+[DSCResource()]
+class WinGetPackage
+{
+    [DscProperty(Key)]
+    [string]$Id
+
+    [DscProperty()]
+    [string]$Version
+
+    [DscProperty()]
+    [string]$Source
+
+    [DscProperty()]
+    [Ensure]$Ensure = [Ensure]::Present
+
+    [DscProperty(NotConfigurable)]
+    [bool] $Installed = $false
+
+    # Get.
+    [WinGetPackage] Get()
+    {
+        Assert-WinGetCommand "Get-WinGetPackage"
+
+        # DSC only validates keys and mandatories in a Set call.
+        if ([string]::IsNullOrWhiteSpace($this.Id))
+        {
+            throw "Id is required"
+        }
+
+        $wingetPackageResource = [WinGetPackage]::new()
+        $wingetPackageResource.Id = $this.Id
+
+        # We can't have CatalogPackage as member of this class because it doesn't have a default constructor.
+        # A call to Invoke-DscResource will fail with
+        # 'ImportClassResourcesFromModule: Exception calling "ImportClassResourcesFromModule" with "4" argument(s): "The DSC resource 'CatalogPackage' has no default constructor."'
+        $catalogPackage = Get-WinGetPackage -Id $this.Id -Exact

```

Comment:
`Exact` is not the correct matching strategy; it requires the case to match.  What we want is `EqualsCaseInsensitive`, but that isn't exposed via PS currently.  We alternate between that and `ContainsCaseInsensitive` in `winget` commands based on whether the command is targeting a single package or multiple, but the PS commands don't carry the actions that are intended during the search phase.

The matching strategy parameter needs to be changed to expose all of the options, probably as a parameter that takes an enumeration value.